File: CaseStudyJavaCore/src/models/Customer.java
package models;

public class Customer {
    private String nameCustomer;
    private String idCard;
    private String birthday;
    private String gender;
    private String cmnd;
    private String phoneNumber;
    private String email;
    private String typeCustomer;
    private String address;
    private Services services;

    public Customer() {
    }

    public Customer(String nameCustomer, String idCard, String birthday, String gender, String cmnd, String phoneNumber,
                    String email, String typeCustomer, String address, Services services) {
        this.nameCustomer = nameCustomer;
        this.idCard = idCard;
        this.birthday = birthday;
        this.gender = gender;
        this.cmnd = cmnd;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.typeCustomer = typeCustomer;
        this.address = address;
        this.services = services;
    }

    public String getNameCustomer() {
        return nameCustomer;
    }

    public void setNameCustomer(String nameCustomer) {
        this.nameCustomer = nameCustomer;
    }

    public String getIdCard() {
        return idCard;
    }

    public void setIdCard(String idCard) {
        this.idCard = idCard;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getCmnd() {
        return cmnd;
    }

    public void setCmnd(String cmnd) {
        this.cmnd = cmnd;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTypeCustomer() {
        return typeCustomer;
    }

    public void setTypeCustomer(String typeCustomer) {
        this.typeCustomer = typeCustomer;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Services getServices() {
        return services;
    }

    public void setServices(Services services) {
        this.services = services;
    }

    public String showInfo() {
        return "Customer information: " + "\n" + "Name: " + this.nameCustomer + "\n" + "IdCard: " + this.idCard+ "\n" + "Birthday: " + this.birthday +
                "\n" + "Gender: " + this.gender + "\n" + "Cmnd: " + this.cmnd + "\n" + "Phone Number: " + this.phoneNumber +
                "\n" + "Email: " + this.email + "\n" + "Type Customer: " + this.typeCustomer + "\n" +
                "Address: " + this.address + "\n" + "Services: " + this.services;
    }

    public int getYearBirthday() {
        String temp = "";
        for (int i = birthday.length() - 4; i < birthday.length(); i++) {
            temp += birthday.charAt(i);
        }
        return new Integer(temp);
    }
}

File: Module1/CaseStudy/Yeucau6/Task1/Customer.js
let Customer = function () {
    this.setName = function (name) {
        this.name = name;
    }
    this.getName = function () {
        return this.name;
    }
    this.setCmnd = function (cmnd) {
        this.cmnd = cmnd;
    }
    this.getCmnd = function () {
        return this.cmnd;
    }
    this.setBirthday = function (birthday) {
        this.birthday = birthday;
    }
    this.getBirthday = function () {
        return this.birthday;
    }
    this.setEmail = function (email) {
        this.email = email;
    }
    this.getEmail = function () {
        return this.email;
    }
    this.setAddress = function (address) {
        this.address = address;
    }
    this.getAddress = function () {
        return this.address;
    }
    this.settypeCustomer = function (typeCustomer) {
        this.typeCustomer = typeCustomer;
    }
    this.gettypeCustomer = function () {
        return this.typeCustomer;
    }
    this.setdiscount = function (discount) {
        this.discount = discount;
    }
    this.getdiscount = function () {
        return this.discount;
    }
    this.setsoluongdikem = function (soluongdikem) {
        this.soluongdikem = soluongdikem;
    }
    this.getsoluongdikem = function () {
        return this.soluongdikem;
    }
    this.setrentDay = function (rentDay) {
        this.rentDay = rentDay;
    }
    this.getrentDay = function () {
        return this.rentDay;
    }
    this.settypeService = function (typeService) {
        this.typeService = typeService;
    }
    this.gettypeService = function () {
        return this.typeService;
    }
    this.settypeRoom = function (typeRoom) {
        this.typeRoom = typeRoom;
    }
    this.gettypeRoom = function () {
        return this.typeRoom;
    }
    this.totalPay = function () {
        let money = 0;
        if (this.typeService === "Villa") {
            money = 500;
        } else if (this.typeService === "House") {
            money = 300;
        } else if (this.typeService === "Room") {
            money = 200;
        }
        let totalPayy = this.rentDay * money * (1 - this.discount / 100);
        return totalPayy;
    }
}